### PR TITLE
install warning stuff complete

### PR DESCRIPTION
**Pull request by ProofOfKeags**

### CODE REVIEW

File: agent/src/Handler/Types/Apps.hs [hunk 1]
```diff
@@ -36,6 +37,7 @@ instance ToJSON AppAvailablePreview where
         , "descriptionShort" .= appAvailablePreviewDescriptionShort
         , "versionInstalled" .= (fst <$> appAvailablePreviewInstallInfo)
         , "status" .= (snd <$> appAvailablePreviewInstallInfo)
+        , "latestVersionTimestamp" .= appAvailablePreviewTimestamp
```
Comment (ProofOfKeags) [inline agent/src/Handler/Types/Apps.hs:40]: @afg419

File: agent/src/Handler/Types/Apps.hs [hunk 2]
```diff
@@ -140,18 +144,21 @@ instance ToJSON AppInstalledFull where
...
     , appVersionInfoReleaseNotes           :: Text
     , appVersionInfoDependencyRequirements :: [Full AppDependencyRequirement]
+    , appVersionInfoInstallWarning         :: Maybe Text
     }
 instance ToJSON AppVersionInfo where
     toJSON AppVersionInfo {..} = object
         [ "version" .= appVersionInfoVersion
         , "releaseNotes" .= appVersionInfoReleaseNotes
         , "serviceRequirements" .= appVersionInfoDependencyRequirements
+        , "installWarning" .= appVersionInfoInstallWarning
```
Comment (ProofOfKeags) [inline agent/src/Handler/Types/Apps.hs:161]: @afg419

File: agent/src/Lib/External/AppManifest.hs [hunk 3]
```diff
@@ -60,25 +60,27 @@ data AppManifest where
...
+        appManifestDescShort        <- o .: "description" >>= (.: "short")
+        appManifestDescLong         <- o .: "description" >>= (.: "long")
+        appManifestReleaseNotes     <- o .: "release-notes"
+        appManifestPortMapping      <- o .: "ports" >>= fmap HM.fromList . traverse parsePortMapping
+        appManifestImageType        <- o .: "image" >>= (.: "type")
+        appManifestMount            <- o .: "mount"
+        appManifestAssets           <- o .: "assets" >>= traverse parseJSON
+        appManifestOnionVersion     <- o .: "hidden-service-version"
+        appManifestDependencies     <- o .:? "dependencies" .!= HM.empty >>= traverse parseDepInfo
+        appManifestUninstallWarning <- o .:? "uninstall-warning"
```
Comment (ProofOfKeags) [inline agent/src/Lib/External/AppManifest.hs:83]: @dr-bonez

File: agent/src/Lib/External/Registry.hs [hunk 4]
```diff
@@ -135,6 +135,7 @@ parseAppData = do
         storeAppVersions         <- ad .: "version-info" >>= \case
             []       -> fail "No Valid Version Info"
             (x : xs) -> pure $ x :| xs
+        storeAppTimestamp <- ad .: "timestamp"
```
Comment (ProofOfKeags) [inline agent/src/Lib/External/Registry.hs:138]: @elvece

File: agent/src/Lib/Types/ServerApp.hs [hunk 5]
```diff
@@ -20,20 +20,23 @@ data StoreApp = StoreApp
...
     deriving (Eq, Show)
 instance Ord StoreAppVersionInfo where
     compare = compare `on` storeAppVersionInfoVersion
 instance FromJSON StoreAppVersionInfo where
     parseJSON = withObject "Store App Version Info" $ \o -> do
-        storeAppVersionInfoVersion      <- o .: "version"
-        storeAppVersionInfoReleaseNotes <- o .: "release-notes"
+        storeAppVersionInfoVersion        <- o .: "version"
+        storeAppVersionInfoReleaseNotes   <- o .: "release-notes"
+        storeAppVersionInfoInstallWarning <- o .: "install-warning"
```
Comment (ProofOfKeags) [inline agent/src/Lib/Types/ServerApp.hs:39]: @dr-bonez

**Comment by ProofOfKeags** [comment]
Tagged the specific lines people should confirm on the API

File: agent/src/Lib/External/AppManifest.hs (hunk 3 above)
Comment (dr-bonez) [reply agent/src/Lib/External/AppManifest.hs:83]: yep

File: agent/src/Lib/Types/ServerApp.hs (hunk 5 above)
Comment (dr-bonez) [reply agent/src/Lib/Types/ServerApp.hs:39]: yep

**Review by dr-bonez — APPROVED** [review]

File: agent/src/Handler/Types/Apps.hs (hunk 2 above)
Comment (afg419) [reply agent/src/Handler/Types/Apps.hs:161]: yep

File: agent/src/Handler/Types/Apps.hs (hunk 1 above)
Comment (afg419) [reply agent/src/Handler/Types/Apps.hs:40]: yep

**Review by afg419 — APPROVED** [review]

File: agent/src/Lib/External/Registry.hs (hunk 4 above)
Comment (elvece) [reply agent/src/Lib/External/Registry.hs:138]: yep

**Review by elvece — APPROVED** [review]

**Pull request merged**